tools/prom-metrics-collector/metrics_json_generator.go

```diff
@@ -21,16 +21,46 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"os"
+	"strings"
+
+	"github.com/kubevirt/monitoring/pkg/metrics/parser"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/monitoring/metrics"
+	metricsparser "github.com/kubevirt/cluster-network-addons-operator/tools/metrics-parser"
 )
 
+// This should be used only for very rare cases where the naming conventions that are explained in the best practices:
+// https://sdk.operatorframework.io/docs/best-practices/observability-best-practices/#metrics-guidelines
+// should be ignored.
+var excludedMetrics = map[string]bool{}
+
 func main() {
-	metricFamilies := ReadMetrics()
+	err := metrics.SetupMetrics()
+	if err != nil {
+		panic(err)
+	}
+
+	metricsList := metrics.ListMetrics()
+	for _, metric := range metricsparser.ReadFromPrometheusCR() {
+		metricsList = append(metricsList, metric)
+	}
+
+	var metricFamilies []parser.Metric
+	for _, m := range metricsList {
+		if excludedMetrics[m.GetOpts().Name] {
+			continue
+		}
+
+		metricFamilies = append(metricFamilies, parser.Metric{
+			Name: m.GetOpts().Name,
+			Help: m.GetOpts().Help,
+			Type: strings.ToUpper(string(m.GetType())),
```

Comment:
Hi @machadovilaca I also had to use: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2583/files#diff-7de3c45a3f3e5eecc9e34ca7a505f4d76146b6cc1cd306a563cf5ad286d8c156R49 in hco since types ended with vec are not expected by the linter (countervec). it didn't fail here since no *vec types. but it will if there will be.

should we update the linter first maybe? or add the line and update in the linter ticket?

Reply:
if we update the linter, maybe its not needed to add that change here

Reply:
Yes that's what I'm asking, fix the linter now or later?

Reply:
It shouldn't be too much work, right? I think we can do it now

Reply:
@sradco WDYT?

Reply:
not too much Just need to make sure to update to the linter new tag in the refactored repos. 
I will work on the linter pr now. thanks

Reply:
can you take a look? https://github.com/kubevirt/monitoring/pull/204